### ISSUE TITLE
IllegalArgumentException with listed example.

### ISSUE THREAD

**Issue by NatKarmios** [scalecube/socketio#43]
```java
import io.netty.buffer.ByteBuf;
import io.scalecube.socketio.Session;
import io.scalecube.socketio.SocketIOAdapter;
import io.scalecube.socketio.SocketIOServer;

public class Test {
    public static void main(String[] args) {
        SocketIOServer echoServer = SocketIOServer.newInstance();
        echoServer.setListener(new SocketIOAdapter() {
            public void onMessage(Session session, ByteBuf message) {
                session.send(message);
            }
        });
        echoServer.start();
    }
}
```

This is listed as your first usage example, however this yields:

```
Exception in thread "main" java.lang.IllegalArgumentException: nThreads: 0 (expected: > 0)
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:72)
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:58)
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:47)
	at io.netty.util.concurrent.DefaultEventExecutorGroup.<init>(DefaultEventExecutorGroup.java:54)
	at io.netty.util.concurrent.DefaultEventExecutorGroup.<init>(DefaultEventExecutorGroup.java:40)
	at io.netty.util.concurrent.DefaultEventExecutorGroup.<init>(DefaultEventExecutorGroup.java:30)
	at io.scalecube.socketio.pipeline.SocketIOChannelInitializer.<init>(SocketIOChannelInitializer.java:109)
	at io.scalecube.socketio.SocketIOServer.start(SocketIOServer.java:122)
	at com.karmios.nat.gnatbot.Test.main(Test.java:19)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
```

**Comment by antonkharenko** [comment]
hm, it seems that Netty changed default behavior for thread count. Previously when you provide zero, it was using `num_of_processors * 2`. I will look into it.

**Comment by antonkharenko** [comment]
@NatKarmios thanks for reporting this issue! It is fixed at PR https://github.com/scalecube/socketio/pull/44 and fix available at new release version `2.3.3`
